api/utils/sshutils/ppk/ppk.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package ppk provides functions implementing conversion between Teleport's native RSA
+// keypairs and PuTTY's PPK format. It also provides functions for working with RFC4251-formatted
+// mpints and strings.
+package ppk
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+
+	"github.com/gravitational/teleport/api/constants"
+	"github.com/gravitational/trace"
+)
+
+// ConvertToPPK takes a regular RSA-formatted keypair and converts it into the PPK file format used by the PuTTY SSH client.
+// The file format is described here: https://the.earth.li/~sgtatham/putty/0.76/htmldoc/AppendixC.html#ppk
+func ConvertToPPK(priv []byte, pub []byte) ([]byte, error) {
```

Comment:
Can you add a `ppk_test.go` with `TestConvertToPPK`? It'd be good to test a happy path along with a couple unhappy paths, like when `ssh-rsa ` is missing.

Reply:
Added in [c47a94b](https://github.com/gravitational/teleport/pull/12733/commits/c47a94b7170c341cdbcc55f388f4131d5fa7ad5a)!